salt/utils/rsax931.py

```diff
@@ -24,43 +24,57 @@
 OPENSSL_INIT_NO_LOAD_CONFIG = 0x00000080
 
 
-def _load_libcrypto():
+def _find_libcrypto():
     """
-    Load OpenSSL libcrypto
+    Find the path (or return the short name) of libcrypto.
     """
     if sys.platform.startswith("win"):
-        # cdll.LoadLibrary on windows requires an 'str' argument
-        return cdll.LoadLibrary(
-            str("libeay32")
-        )  # future lint: disable=blacklisted-function
+        lib = str("libeay32")
     elif getattr(sys, "frozen", False) and salt.utils.platform.is_smartos():
-        return cdll.LoadLibrary(
-            glob.glob(os.path.join(os.path.dirname(sys.executable), "libcrypto.so*"))[0]
-        )
+        lib = glob.glob(os.path.join(os.path.dirname(sys.executable), "libcrypto.so*"))
+        lib = lib[0] if lib else None
     else:
-        lib = find_library("crypto")
-        if not lib and salt.utils.platform.is_sunos():
-            # Solaris-like distribution that use pkgsrc have
-            # libraries in a non standard location.
-            # (SmartOS, OmniOS, OpenIndiana, ...)
-            # This could be /opt/tools/lib (Global Zone)
-            # or /opt/local/lib (non-Global Zone), thus the
-            # two checks below
-            lib = glob.glob("/opt/local/lib/libcrypto.so*") + glob.glob(
-                "/opt/tools/lib/libcrypto.so*"
-            )
-            lib = lib[0] if len(lib) > 0 else None
-        if not lib and salt.utils.platform.is_aix():
-            if os.path.isdir("/opt/salt/lib"):
-                # preference for Salt installed fileset
-                lib = glob.glob("/opt/salt/lib/libcrypto.so*")
-                lib = lib[0] if len(lib) > 0 else None
+        lib = ctypes.util.find_library("crypto")
+        if not lib:
+            if salt.utils.platform.is_sunos():
+                # Solaris-like distribution that use pkgsrc have libraries
+                # in a non standard location.
+                # (SmartOS, OmniOS, OpenIndiana, ...)
+                # This could be /opt/tools/lib (Global Zone) or
+                # /opt/local/lib (non-Global Zone), thus the two checks
+                # below
+                lib = glob.glob("/opt/local/lib/libcrypto.so*")
+                lib = lib or glob.glob("/opt/tools/lib/libcrypto.so*")
+                lib = lib[0] if lib else None
+            elif salt.utils.platform.is_aix():
+                if os.path.isdir("/opt/salt/lib"):
+                    # preference for Salt installed fileset
+                    lib = glob.glob("/opt/salt/lib/libcrypto.so*")
+                else:
+                    lib = glob.glob("/opt/freeware/lib/libcrypto.so*")
+                lib = lib[0] if lib else None
+        elif salt.utils.platform.is_darwin():
```

Comment:
@kaorihinata this appears to not be indented enough and never gets run.

Reply:
if this gets moved to right one block and includes your changes above it will run on macOS Big Sur.

Reply:
Along with the changes in #57607

Reply:
The indentation level is not the mistake, I forgot part of the bug. It's going to need to be rewritten a bit. If `lib` is not `None`, but we are on darwin then it's most likely found a non-versioned lib in a system location which is the bug. This needs to be run if `not lib`. Lemme change that.